src/analytics/lambdas/user-segments-workflow/segment-job-status.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { StatusString } from '@aws-sdk/client-redshift-data';
+import { DynamoDBDocumentClient, UpdateCommand } from '@aws-sdk/lib-dynamodb';
+import { ExecuteSegmentQueryOutput } from './execute-segment-query';
+import { logger } from '../../../common/powertools';
+import { aws_sdk_client_common_config } from '../../../common/sdk-client-config';
+import { parseDynamoDBTableARN } from '../../../common/utils';
+import { handleBackoffTimeInfo } from '../../../common/workflow';
+import { SegmentJobStatus } from '../../private/segments/segments-model';
+import { describeStatement, getRedshiftClient } from '../redshift-data';
+
+export interface SegmentJobStatusEvent extends ExecuteSegmentQueryOutput {
+  jobStatus?: SegmentJobStatus;
+}
+
+const { ddbRegion, ddbTableName } = parseDynamoDBTableARN(process.env.CLICKSTREAM_METADATA_DDB_ARN!);
+const ddbClient = new DynamoDBClient({
+  ...aws_sdk_client_common_config,
+  region: ddbRegion,
+});
+const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);
+
+const redshiftClient = getRedshiftClient(process.env.REDSHIFT_DATA_API_ROLE!);
+
+const _handler = async (event: SegmentJobStatusEvent) => {
```

Comment:
add verbose logs for troubleshooting potential issues

Reply:
done